### PR TITLE
changed member input form to be a modal using daisyUI pending making it work

### PR DESCRIPTION
**Pull request by malex37**

### CODE REVIEW

File: src/components/RegisterMember.tsx [hunk 1]
```diff
@@ -1,7 +1,8 @@
-import React, { useState } from 'react';
+import React, { useRef, useState } from 'react';
```
Comment (iamskok) [inline src/components/RegisterMember.tsx:1]: The linter didn't yell at you?

File: src/components/Team.tsx [hunk 2]
```diff
@@ -13,26 +15,28 @@ const Team = () => {
...
-    logger.debug(`Retrieved teamName ${teamNameString}`);
-    if (teamNameString) {
-      
-      const fetchTeamMembers = async () => {
-        return await DbProvider.getTeam(teamNameString);
-      };
-      fetchTeamMembers().then(data => {
-        setState({teamName: teamNameString, members: data.members});
-      });
+    if (!init) {
```
Comment (iamskok) [inline src/components/Team.tsx:21]: Do we need `init` if it's never set to `true`? 

I think we should also consider adding a new task for the  "sad" path. What if `team` is not found? Should we render some kind of an error message?

File: src/components/display/UserDashboard.tsx [hunk 3]
```diff
@@ -12,19 +12,22 @@ import SprintStatsRow from './SprintStatsRow';
...
-    }; 
-    fetchData().then(data => {
-      setState(data);
-    }).catch(error => {
-      logger.error(`Error fetching data ${JSON.stringify(error)}`);
-    });
-  }, [state]);
+    };
+    if (!initiated) { 
+      initiated = true;
```
Comment (iamskok) [inline src/components/display/UserDashboard.tsx:23]: Did you take this pattern from React docs?

**Review by iamskok — APPROVED** [review]

File: src/components/display/UserDashboard.tsx (hunk 3 above)
Comment (malex37) [reply src/components/display/UserDashboard.tsx:23]: Yeah I did

File: src/components/Team.tsx (hunk 2 above)
Comment (malex37) [reply src/components/Team.tsx:21]: I might've missed it it should be assigned, I'll send an update.  And yeah we should probably render an error

**Pull request merged**